commands/cli_nukeallresources_test.go

```diff
@@ -0,0 +1,547 @@
+	if ok {
+		ec2Instances := aws.EC2Instances{}
+		ec2Instances.InstanceIds = []string{resourceIdentifier}
+		resourcesInRegion.Resources = append(resourcesInRegion.Resources, ec2Instances)
+	}
+
+	if len(resourcesInRegion.Resources) > 0 {
+		nukeAccount.Resources[targetRegion] = resourcesInRegion
+		return aws.NukeAllResourcesArgs{
+			Account:                &nukeAccount,
+			Regions:                []string{targetRegion},
+			IgnoreErrResourceTypes: ignoreErrResourceTypes,
+			RegionSessionMap:       map[string]*session.Session{targetRegion: awsSession},
+		}
+	}
+	return aws.NukeAllResourcesArgs{}
+}
+
+// TestNukeAllResourcesAllPerms tests deletion of test S3 and EC2 instance with user's
+// default AWS session
+func testNukeAllResources(t *testing.T, awsParams aws.CloudNukeAWSParams, assumeRoleArn string, args testNukeAllResourcesArgs) {
+	// Generate test bucket and test instance name
+	bucketName := aws.S3GenBucketName()
+	ec2InstanceName := "cloud-nuke-test-" + util.UniqueID()
+
+	resourceTypeIdentifierMap := map[string]string{}
+	resourceTypes := []string{}
+	tags := []map[string]string{{"Key": "cloud-nuke-test", "Value": "true"}}
+
+	// Create test bucket and test instance
+	resourceTypeIdentifierMap, err := createTestResources(
+		awsParams.AWSSession,
+		[]createTestResourcesArgs{
+			{resourceType: "s3", name: bucketName, tags: tags},
+			{resourceType: "ec2", name: ec2InstanceName, tags: tags},
+		},
+	)
+
+	// Ensure test bucket + test EC2 instance gets deleted at the end of the test
+	// Add defer before checking error to handle the case where partial resources get
+	// created and must be destroyed
+	defer aws.NukeAllResources(buildNukeAllResourcesArgs(awsParams.AWSSession, resourceTypeIdentifierMap, awsParams.Region, args.ignoreErrResourceTypes))
+
+	require.NoError(t, err, "Failed to create test resources")
+
+	for resourceType := range resourceTypeIdentifierMap {
+		resourceTypes = append(resourceTypes, resourceType)
+	}
+
+	// Assume role
+	var roleSession *session.Session
+	if assumeRoleArn == "" {
+		roleSession = awsParams.AWSSession
+	} else {
+		roleSession, err = aws.AssumeIAMRole(assumeRoleArn, awsParams.Region)
+		require.NoErrorf(t, err, "Failed to assume role - %s - %s", assumeRoleArn, err)
+	}
+
+	excludeAfter, err := parseDurationParam("0s")
+	require.NoError(t, err, "parseDurationParam failed")
+
+	// Do assumed role get for existing resources of type EC2 and S3
+	account, err := aws.GetAllResources(
+		aws.GetAllResourcesArgs{
+			TargetRegions:          []string{awsParams.Region},
+			ExcludeAfter:           *excludeAfter,
+			NukeResourceTypes:      resourceTypes,
+			IgnoreErrResourceTypes: args.ignoreErrResourceTypes,
+			RegionSessionMap:       map[string]*session.Session{awsParams.Region: roleSession},
+		},
+	)
+	// Validate that get call should fail if specified
+	if args.getResourcesShouldFail {
+		require.Error(t, err, "GetAllResources did not fail")
```

Comment:
Hm, prob a good idea to check _which_ error you get here!

Reply:
Given the current codebase, I don't think we can check. For example `GetAllResources` calls get for all resource types, each of which returns `errors.WithStackTrace(err)` - how do we check specific error type in this case?